File: backend/app/schemas/tenant.py
"""
用户 Schemas
"""

from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class TenantCreate(BaseModel):
    """创建用户"""
    name: str = Field(..., max_length=200, description="用户名称")
    code: Optional[str] = Field(default=None, max_length=50, description="用户编码 (留空则按新增顺序自动生成)")
    account: str = Field(..., max_length=100, description="登录账号")
    password: str = Field(..., min_length=6, max_length=100, description="登录密码")
    contact_name: Optional[str] = None
    contact_phone: Optional[str] = None
    expires_at: Optional[datetime] = None
    max_users: int = Field(default=5, ge=1, le=1000)
    max_video_projects: int = Field(default=50, ge=1, le=10000)
    config: Optional[dict] = None


class TenantUpdate(BaseModel):
    """更新用户"""
    name: Optional[str] = None
    contact_name: Optional[str] = None
    contact_phone: Optional[str] = None
    expires_at: Optional[datetime] = None
    max_users: Optional[int] = None
    max_video_projects: Optional[int] = None
    config: Optional[dict] = None
    status: Optional[int] = Field(default=None, ge=-1, le=1)


class TenantResponse(BaseModel):
    """用户响应"""
    id: int
    name: str
    code: str
    account: str
    contact_name: Optional[str] = None
    contact_phone: Optional[str] = None
    expires_at: Optional[datetime] = None
    status: int
    max_users: int
    max_video_projects: int
    config: Optional[dict] = None
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class TenantListResponse(BaseModel):
    items: list[TenantResponse]
    total: int


# ========== 用户登录 ==========
class TenantLogin(BaseModel):
    """用户登录"""
    account: str
    password: str


class TenantLoginResponse(BaseModel):
    """用户登录响应"""
    token: str
    tenant: TenantResponse


# ========== 子账号管理 ==========
class TenantUserCreate(BaseModel):
    """创建子账号"""
    phone: str = Field(..., max_length=20)
    password: str = Field(..., min_length=6, max_length=100)
    nickname: Optional[str] = None
    role: str = Field(default="anchor", description="anchor=主播, tenant_admin=用户管理员")

File: frontend/src/api/tenant.ts
import request from '@/utils/request'
import type { Tenant, TenantListResponse, TenantLoginResponse } from '@/types/user'

// 渠道商登录
export function tenantLogin(account: string, password: string) {
  return request.post<unknown, TenantLoginResponse>('/tenants/login', {
    account,
    password,
  })
}

// 获取渠道商列表（总部管理员）
export function listTenants(skip = 0, limit = 100) {
  return request.get<unknown, TenantListResponse>('/tenants/', {
    params: { skip, limit },
  })
}

// 创建渠道商（总部管理员）
export function createTenant(data: {
  name: string
  code: string
  account: string
  password: string
  contact_name?: string
  contact_phone?: string
  expires_at?: string
  max_users?: number
  max_video_projects?: number
}) {
  return request.post<unknown, Tenant>('/tenants/', data)
}

// 更新渠道商（总部管理员）
export function updateTenant(id: number, data: Partial<{
  name: string
  contact_name: string
  contact_phone: string
  expires_at: string
  max_users: number
  max_video_projects: number
  status: number
}>) {
  return request.put<unknown, Tenant>(`/tenants/${id}`, data)
}

// 停用渠道商
export function deleteTenant(id: number) {
  return request.delete<unknown, { message: string }>(`/tenants/${id}`)
}

// 获取渠道商下的用户列表
export function listTenantUsers(tenantId: number) {
  return request.get<unknown, { items: Array<{ id: number; phone: string; nickname: string; role: string; status: number }>; total: number }>(`/tenants/${tenantId}/users`)
}

// 创建渠道用户
export function createTenantUser(tenantId: number, data: {
  phone: string
  password: string
  nickname?: string
  role?: string
}) {
  return request.post<unknown, { message: string; user_id: number }>(`/tenants/${tenantId}/users`, data)
}
